File: tests_deps/src/test/java/com/github/klee0kai/stone/test/lists/cachecontrol/gc/BumperGcResusableTests.java
package com.github.klee0kai.stone.test.lists.cachecontrol.gc;

import com.github.klee0kai.stone.Stone;
import com.github.klee0kai.stone.closed.types.ListUtils;
import com.github.klee0kai.test.car.di.cachecontrol.gc.CarGcComponent;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotEquals;

public class BumperGcResusableTests {

    @Test
    public void gcAllTest() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcAll();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertNotEquals(bumperSoftUids1, bumperSoftUids2);
        assertNotEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcWeakTest() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcWeak();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcSoftTest() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcSoft();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertNotEquals(bumperSoftUids1, bumperSoftUids2);
        assertEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcStrongTest() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcStrong();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertNotEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcBumpers() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcBumpers();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertNotEquals(bumperSoftUids1, bumperSoftUids2);
        assertNotEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcRedBumpers() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcRedBumpers();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertNotEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcRedBumpers2() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcRedBumpers2();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertNotEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcWheelsTest() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcWheels();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertEquals(bumperStrongUids1, bumperStrongUids2);
    }

    @Test
    public void gcNothing() {
        // Given
        CarGcComponent DI = Stone.createComponent(CarGcComponent.class);
        List<String> bumperFactoryUids1 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids1 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids1 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids1 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        //When
        DI.gcNothing();
        List<String> bumperFactoryUids2 = ListUtils.format(DI.bumpersModule().bumperFactory(), it -> it.uuid);
        List<String> bumperWeakUids2 = ListUtils.format(DI.bumpersModule().bumperWeak(), it -> it.uuid);
        List<String> bumperSoftUids2 = ListUtils.format(DI.bumpersModule().bumperSoft(), it -> it.uuid);
        List<String> bumperStrongUids2 = ListUtils.format(DI.bumpersModule().bumperStrong(), it -> it.uuid);

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2);
        assertNotEquals(bumperWeakUids1, bumperWeakUids2);
        assertEquals(bumperSoftUids1, bumperSoftUids2);
        assertEquals(bumperStrongUids1, bumperStrongUids2);
    }

}

File: tests_wraps_kotlin/src/test/kotlin/com/github/klee0kai/stone/test/car/cachecontrol/gc/BumperGcResusableTests.kt
package com.github.klee0kai.stone.test.car.cachecontrol.gc

import com.github.klee0kai.stone.Stone
import com.github.klee0kai.test.car.di.cachecontrol.gc.CarGcComponent
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Test

class BumperGcResusableTests {
    @Test
    fun gcAllTest() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcAll()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertNotEquals(bumperSoftUids1, bumperSoftUids2)
        assertNotEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcWeakTest() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcWeak()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }


        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcSoftTest() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcSoft()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertNotEquals(bumperSoftUids1, bumperSoftUids2)
        assertEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcStrongTest() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcStrong()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertNotEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcBumpers() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcBumpers()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertNotEquals(bumperSoftUids1, bumperSoftUids2)
        assertNotEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcRedBumpers() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcRedBumpers()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertNotEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcRedBumpers2() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcRedBumpers2()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertNotEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcWheelsTest() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcWheels()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertEquals(bumperStrongUids1, bumperStrongUids2)
    }

    @Test
    fun gcNothing() {
        // Given
        val DI = Stone.createComponent(CarGcComponent::class.java)
        val bumperFactoryUids1 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids1 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids1 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids1 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        //When
        DI.gcNothing()
        val bumperFactoryUids2 = DI.bumpersModule()!!.bumperFactory().map { it.uuid }
        val bumperWeakUids2 = DI.bumpersModule()!!.bumperWeak().map { it.uuid }
        val bumperSoftUids2 = DI.bumpersModule()!!.bumperSoft().map { it.uuid }
        val bumperStrongUids2 = DI.bumpersModule()!!.bumperStrong().map { it.uuid }

        // Then
        assertNotEquals(bumperFactoryUids1, bumperFactoryUids2)
        assertNotEquals(bumperWeakUids1, bumperWeakUids2)
        assertEquals(bumperSoftUids1, bumperSoftUids2)
        assertEquals(bumperStrongUids1, bumperStrongUids2)
    }
}
